onnxruntime/core/graph/model.h

```diff
@@ -27,6 +27,25 @@ struct Model;
 typedef std::unordered_map<std::string, std::string> ModelMetaData;
 using IOnnxRuntimeOpSchemaRegistryList = std::list<std::shared_ptr<IOnnxRuntimeOpSchemaCollection>>;
 
+// Options to configure Model.
+struct ModelOptions {
+  bool allow_released_opsets_only;
+  // If true, all inconsistencies encountered during shape and type inference
+  // will be exposed to the caller as failures. If false, in some cases
+  // warnings will be logged but processing will continue and no error will
+  // be returned.
+  bool strict_shape_type_inference;
+
+  ModelOptions(bool allow_released_opsets_only, bool strict_shape_type_inference)
+      : allow_released_opsets_only(allow_released_opsets_only), strict_shape_type_inference(strict_shape_type_inference) {}
+
+  ModelOptions() : ModelOptions(true, false) {}
+};
+
+namespace {
+static ModelOptions default_model_options;
+}  // namespace
```

Comment:
Does this get inlined into every .cc file that includes this header? 

Do we actually need a static variable for it vs. creating on demand as a default arg? e.g.

```
  explicit Model(const std::string& graph_name,
                 bool is_onnx_domain_only,
                 const logging::Logger& logger,
                 const ModelOptions& options = {})
```

Reply:
Done.